scripts/purge-platform-pkgs.py

```diff
@@ -0,0 +1,86 @@
+#!/usr/bin/env python
+"""Strip unwanted packages out of pip-compiled requirements txt files.
+
+TODO:
+- support packages with multiple via's
+- accept purge package details via --package argument instead of global dict
+"""
+import argparse
+import sys
+from collections import deque
+
+
+PACKAGES_TO_PURGE = {
+    # package, via
+    "appnope": "ipython",
+}
+
+
+def main(files, packages=PACKAGES_TO_PURGE):
+    for file in files:
+        # Ideally these things are less than GiB's in size, because we're gonna
+        # have a few copies in memory :)
+        lines = list(file)
+        purged = list(purge_packages(lines, packages))
+        if file.fileno() == 0:  # stdin
+            # always write these lines to stdout, regardless of purge
+            rewrite_lines(sys.stdout, purged)
```

Comment:
I assume since this is here that one way to use this script is to pipe a file to it as in `cat requirements.txt | scripts/purge-platform-pkgs.py -`. If that is all correct, cool. If not, can you explain what this is for?

Reply:
That is correct.